### ISSUE TITLE
overflow-wrap-break-word-fit-content-001.html

### ISSUE THREAD

**Issue by frivoal** [web-platform-tests/wpt#14549]
There are three problems with [css/css-text/overflow-wrap/overflow-wrap-break-word-fit-content-001.html](https://github.com/web-platform-tests/wpt/blob/master/css/css-text/overflow-wrap/overflow-wrap-break-word-fit-content-001.html), and as I am not 100% sure what it is meant to be testing, I'd rather ask first before attempting to fix it. @kojiishi, you're listed as the author of this test, so maybe you can help.

* The filename and the `<title>` suggest this test has something to do with `fit-content`, but there doesn't seem to be anything in the test about fit content. Is this a (copy&paste?) mistake when making the test where you forgot to rename it? Did you mean to write a different test? Is it actually related to `content-fit` somehow that I don't understand?

* If the assertion "The 'overflow-wrap: break-word' must not break surrogate pairs" is the test purpose, a narrow `width` should be specified on the div, e.g., `.test { width: 1px; }`, so that something is actually trying to cause a line break.

* Still assuming that the assertion does reflect the tests's purpose, the file is not encoded in UTF-16, and therefore does not actually contain surrogate pairs. The character used `&#x1D70B;` is one that would be encoded using surrogate pairs if the encoding was UTF-16, but it isn't. That doesn't make the test wrong: since it is checking that nothing weird happens, the test will correctly pass. But it may not really be testing anything. I suppose that the test is assuming that the browser's internal representation is UTF-16, and that this would therefore be converted into surrogate pairs, and that's why it might fail if the browser handled this incorrectly. I don't think any spec mandates the use of UTF-16 internally, so it is an implementation detail. But I suppose it is a plausible bug. Is that indeed what the test is trying to check? If so, that should be mentioned in the assertion.

In conclusion, if I understand correctly what the test is trying to do, we should:
* change the title and the file name
* add `.test { width: 1px; }`
* `s/surrogate pairs/characters that would be represented by surrogate pairs in the UA's internal representation was UTF-16/` in the assert.

Or maybe I just don't understand this test.

**Comment by kojiishi** [comment]
The test and title seems to match to me, `inline-block` is `fit-content` in the current our terminology, no? If we add `width: 1px`, that'd be a test for `min-content` IIUC, but if this seems wrong, please go ahead and fix.

Did we lose all the git history? I can't see any from github, but from Blink repo, I found this is a test for [crbug.com/574577](https://crbug.com/574577). It looks like test matches to the bug report, so if `inline-block` is not `fit-content`, it should be the title to fix.

For the last point, IIUC, "surrogate pairs" means what you explained regardless of encoding. If you think this is wrong, please go ahead and fix. Checking unicode.org, maybe [surrogates](http://unicode.org/faq/utf_bom.html#utf16-2) is more correct? But maybe less easier to understand. I don't have prereference, if this was hard to understand for you, I'd be happy to take any suggestions.

**Comment by frivoal** [comment]
> The test and title seems to match to me, inline-block is fit-content in the current our terminology, no?

You're right, I somehow missed this. Sorry. Closing.

>>    I think just going with "non-BMP characters" is fine here.

> That is fine with me too

Good suggestion, thanks.
